src/sentry/api/endpoints/project_symbol_sources.py

```diff
@@ -0,0 +1,245 @@
+from uuid import uuid4
+
+from drf_spectacular.utils import OpenApiParameter, extend_schema
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.apidocs.constants import RESPONSE_BAD_REQUEST, RESPONSE_FORBIDDEN, RESPONSE_NOT_FOUND
+from sentry.apidocs.examples.project_examples import ProjectExamples
+from sentry.apidocs.parameters import GlobalParams
+from sentry.lang.native.sources import (
+    InvalidSourcesError,
+    backfill_source,
+    parse_sources,
+    redact_source_secrets,
+    validate_sources,
+)
+from sentry.models import Project
+from sentry.utils import json
+
+
+@extend_schema(tags=["Projects"])
+@region_silo_endpoint
+class ProjectSymbolSourcesEndpoint(ProjectEndpoint):
+    owner = ApiOwner.OWNERS_NATIVE
+    publish_status = {
+        "GET": ApiPublishStatus.PUBLIC,
+        "DELETE": ApiPublishStatus.PUBLIC,
+        "POST": ApiPublishStatus.PUBLIC,
+        "PUT": ApiPublishStatus.PUBLIC,
+    }
+
+    @extend_schema(
+        operation_id="Retrieve a Project's symbol sources",
+        parameters=[
+            GlobalParams.ORG_SLUG,
+            GlobalParams.PROJECT_SLUG,
+            OpenApiParameter(
+                name="id",
+                description="The id of the source to look up.",
+                required=False,
+                type=str,
+                location="query",
+            ),
```

Comment:
This param should be put in this [file](https://github.com/getsentry/sentry/blob/master/src/sentry/apidocs/parameters.py) under the ProjectParams class. It should be a static class function b/c you're using it multiple times on this page w/ different descriptions.

Reply:
I assume the function should get a flag for whether the parameter is required?

Reply:
I've been creating the functions as needed so if a parameter shows up as optional on one endpoint but not the other, then I include a flag. It's up to you though, either way is ok

Reply:
Nice, that's what I ended up doing.